src/pages/index.js

```diff
@@ -0,0 +1,302 @@
+const profileForm = document.querySelector(".modal__form"); // edit form
+const nameOnPage = document.querySelector("#profile-name");
+const descriptionOnPage = document.querySelector("#profile-description");
+const profileExitButton = profileModal.querySelector("#exit-button"); // exit button for both modals
+const profileSubmitBtn = profileModal.querySelector(".modal__submit-btn");
+
+const cardSubmitButton = newPostModal.querySelector(".modal__submit-btn");
+const profileAddButton = document.querySelector(".profile__add-btn");
+const modalImageForm = newPostModal.querySelector(".modal__form"); // image Form
+const imageLink = newPostModal.querySelector("#link-input");
+const caption = newPostModal.querySelector("#caption-input");
+const imageSaveButton = newPostModal.querySelector("#save-image");
+const newModalCloseButton = newPostModal.querySelector(".modal__exit-btn");
+
+const previewModal = document.querySelector("#preview-modal");
+const previewImage = previewModal.querySelector(".modal__image");
+const previewCaption = previewModal.querySelector(".modal__caption");
+const previewModalCloseBtn = previewModal.querySelector(".modal__close");
+
+// avatar modal
+const avatarModalBtn = document.querySelector(".profile__avatar-btn");
+const avatarModal = document.querySelector("#edit-avatar-modal");
+const avatarCloseBtn = document.querySelector("#avatar-exit-button");
+const avatarSubmitButton = document.querySelector("#avatar-submit-button");
+const avatarForm = document.querySelector("#edit-avatar-form");
+const avatarImg = document.querySelector(".profile__avatar");
+const avatarInput = document.querySelector("#avatar-input");
+
+//delete card Modal
+const deleteCardModal = document.querySelector("#delete-card-modal");
+const deleteForm = document.querySelector("#delete-card-form");
+const deleteExitBtn = document.querySelector(".modal__delete-exit-btn");
+const deleteCancel = document.querySelector("#delete-card-cancel");
+const deleteSubmitBtn = document.querySelector("#delete-submit-button");
+
+let selectedCard;
+let selectedCardId;
+
+newModalCloseButton.addEventListener("click", () => {
+  closeModal(newPostModal);
+});
+
+profileAddButton.addEventListener("click", () => {
+  openModal(newPostModal);
+});
+
+profileExitButton.addEventListener("click", () => {
+  closeModal(profileModal);
+});
+
+avatarModalBtn.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+
+avatarCloseBtn.addEventListener("click", () => {
+  closeModal(avatarModal);
+});
+
+deleteExitBtn.addEventListener("click", () => {
+  closeModal(deleteCardModal);
+});
+
+function handleAvatarSubmit(evt) {
+  // form for Avatar changing
+  evt.preventDefault();
+  setButtonText(avatarSubmitButton, true);
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      avatarImg.src = data.avatar;
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(avatarSubmitButton, false);
+    });
+  closeModal(avatarModal);
```

Comment:
You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**